lib/awful/ewmh.lua

```diff
@@ -42,18 +42,13 @@ local function geometry_change(window)
     if geometry_change_lock then return end
     geometry_change_lock = true
 
-    -- Fix up the geometry in case this window needs to cover the whole screen.
-    local bw = window.border_width or 0
-    local g = window.screen.workarea
-    if window.maximized_vertical or window.maximized then
-        window:geometry { height = g.height - 2*bw, y = g.y }
-    end
-    if window.maximized_horizontal or window.maximized then
-        window:geometry { width = g.width - 2*bw, x = g.x }
-    end
-    if window.fullscreen then
-        window.border_width = 0
-        window:geometry(window.screen.geometry)
+    for prop in ipairs {
+        "fullscreen", "maximized", "maximized_vertical", "maximized_horizontal"
+    } do
+        if window[prop] then
+            window:emit_signal("request::geometry", prop, {})
```

Comment:
Should the last argument somehow indicate that this is just re-applying the way things should be?